src/test_l3_plat_power.c

```diff
@@ -0,0 +1,346 @@
+const static ut_control_keyStringMapping_t plat_source_types_mapTable [] = {
+  { "PWRMGR_WAKEUPSRC_VOICE",               (int32_t)PWRMGR_WAKEUPSRC_VOICE     },
+  { "PWRMGR_WAKEUPSRC_PRESENCE_DETECTION",  (int32_t)PWRMGR_WAKEUPSRC_PRESENCE_DETECTION      },
+  { "PWRMGR_WAKEUPSRC_BLUETOOTH",           (int32_t)PWRMGR_WAKEUPSRC_BLUETOOTH     },
+  { "PWRMGR_WAKEUPSRC_WIFI",                (int32_t)PWRMGR_WAKEUPSRC_WIFI   },
+  { "PWRMGR_WAKEUPSRC_IR",                  (int32_t)PWRMGR_WAKEUPSRC_IR  },
+  { "PWRMGR_WAKEUPSRC_POWER_KEY",           (int32_t)PWRMGR_WAKEUPSRC_POWER_KEY },
+  { "PWRMGR_WAKEUPSRC_TIMER",               (int32_t)PWRMGR_WAKEUPSRC_TIMER },
+  { "PWRMGR_WAKEUPSRC_CEC",                 (int32_t)PWRMGR_WAKEUPSRC_CEC },
+  { "PWRMGR_WAKEUPSRC_LAN",                 (int32_t)PWRMGR_WAKEUPSRC_LAN },
+  { "PWRMGR_WAKEUPSRC_MAX",                 (int32_t)PWRMGR_WAKEUPSRC_MAX       },
+  {  NULL, -1 }
+};
+
+
+/* pmStatus_t */
+const static ut_control_keyStringMapping_t pmStatus_mapTable [] = {
+  { "PWRMGR_SUCCESS",                   (int32_t)PWRMGR_SUCCESS                     },
+  { "PWRMGR_INVALID_ARGUMENT",          (int32_t)PWRMGR_INVALID_ARGUMENT                 },
+  { "PWRMGR_ALREADY_INITIALIZED",       (int32_t)PWRMGR_ALREADY_INITIALIZED           },
+  { "PWRMGR_NOT_INITIALIZED",           (int32_t)PWRMGR_NOT_INITIALIZED           },
+  { "PWRMGR_INIT_FAILURE",              (int32_t)PWRMGR_INIT_FAILURE     },
+  { "PWRMGR_SET_FAILURE",               (int32_t)PWRMGR_SET_FAILURE         },
+  { "PWRMGR_GET_FAILURE",               (int32_t)PWRMGR_GET_FAILURE },
+  { "PWRMGR_OPERATION_NOT_SUPPORTED",   (int32_t)PWRMGR_OPERATION_NOT_SUPPORTED  },
+  { "PWRMGR_TERM_FAILURE",              (int32_t)PWRMGR_TERM_FAILURE        },
+  { "PWRMGR_MAX",                       (int32_t)PWRMGR_MAX                                           },
+  {  NULL, -1 }
+};
+
+
+/**
+ * @brief This function clears the stdin buffer.
+ *
+ * This function clears the stdin buffer.
+ */
+void readAndDiscardRestOfLine(FILE* in)
+{
+   int c;
+   while ( (c = fgetc(in)) != EOF && c != '\n');
+}
+
+/**
+* @brief Initialization of the HAL Power Manager Module
+*
+* This test provides a scope to open the HAL Power Manager module.
+
+*
+* **Test Group ID:** 03@n
+*
+* **Test Case ID:** 001@n
+*
+* **Pre-Conditions:** None@n
+*
+* **Dependencies:** None@n
+*
+* **User Interaction:** @n
+* User or Automation tool should select the Test 1 to before running any test.
+*
+*/
+
+void test_l3_power_manager_hal_Init(void)
+{
+   gTestID = 1;
+   pmStatus_t status = PWRMGR_SUCCESS ;
+
+   UT_LOG_INFO("In %s [%02d%03d]\n", __FUNCTION__, gTestGroup, gTestID);
+
+   // Step 1: Call PLAT_INIT()
+   status = PLAT_INIT();
+   UT_LOG_INFO("Result PLAT_INIT: pmStatus_t:[%s]",
+                UT_Control_GetMapString(pmStatus_mapTable, status));
+
+   UT_LOG_INFO("Out %s\n", __FUNCTION__);
+}
+
+/**
+* @brief Initialization of the HAL Power Manager Module
```

Comment:
Brief Description for this function looks to be wrong. Please check and change it.

Reply:
Updated as requested.